devito/propagator.py

```diff
@@ -178,8 +212,17 @@ def generate_space_loops_blocking(self, loop_body):
         ivdep = True
         remainder = False
         orig_loop_body = loop_body
+
+        inner_most_dim_passed = False  # used in oder to avoid at inner most dimension
+
         for spc_var, block_size in reversed(zip(list(self.space_dims), self.block_sizes)):
             dim_var = str(self._var_map[spc_var])
+
+            if self.auto_tune and inner_most_dim_passed:  # change block size into var
```

Comment:
Throughout this file, `block_size` is somewhere an integer and somewhere a string (with the same name). While there is a number set inside `block_size`, you choose to ignore it and only use a different variable called `tune_b_size`. 
This is highly confusing and might lead to huge problems later.